File: abc087-b/java/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            int coin500 = scanner.nextInt();
            int coin100 = scanner.nextInt();
            int coin50 = scanner.nextInt();
            int total = scanner.nextInt();

            int count = 0;
            for (int i = 0; i <= coin500; i++) {
                for (int j = 0; j <= coin100; j++) {
                    for (int k = 0; k <= coin50; k++) {
                        if (i * 500 + j * 100 + k * 50 == total) {
                            count++;
                        }
                    }
                }
            }

            System.out.println(count);
        }
    }
}

File: abc087-b/js/main.js
function Main(input) {
    const [coin500, coin100, coin50, total] = input.split("\n").map(Number);
    let count = 0;
    for (let i = 0; i <= coin500; i++) {
        for (let j = 0; j <= coin100; j++) {
            for (let k = 0; k <= coin50; k++) {
                if (500 * i + 100 * j + 50 * k === total) {
                    count++;
                }
            }
        }
    }
    console.log(count);
}

Main(require("fs").readFileSync("/dev/stdin", "utf8"));
